src/nodes.coffee

```diff
@@ -1186,7 +1192,7 @@ exports.Obj = class Obj extends Base
     for prop in @properties
       prop = prop.value if prop instanceof Assign and prop.context is 'object'
       prop = prop.unwrapAll()
-      prop.eachName iterator
+      prop.eachName iterator if prop.eachName?
```

Comment:
I expect this is to do with your above point that `Param`s aren't compiled as `Assign`s by default.